### PR TITLE
All 5 WCF facades now build and run their related unit tests.

### PR DESCRIPTION
**Pull request by StephenBonikowsky**
* I was not able to verify for UAP as my dev environment has some issue.
* The test projects also use project references to the common code projects.

### CODE REVIEW

**Comment by StephenBonikowsky** [comment]
@dagood I need to clean this up a bit more, like I don't think I needed to put in the pkgprojects back in yet, I made the common projects reference ref projects instead (I hope that is correct).

Was just wondering if you could try running these against UAP, they work for me against netcoreapp but I still can't run against UAP.

File: external/test-runtime/XUnit.Runtime.depproj [hunk 1]
```diff
@@ -78,11 +78,21 @@
...
     <PackageToInclude Include="xunit.extensibility.execution"/>
     <PackageToInclude Include="xunit.runner.utility"/>
+    <PackageToInclude Include="xunit.Sdk"/>
     <PackageToInclude Include="Microsoft.xunit.netcore.extensions"/>
     <PackageToInclude Include="Newtonsoft.Json"/>
     <PackageToInclude Include="$(XUnitRunnerPackageId)" />
   </ItemGroup>
 
+  <PropertyGroup Condition="'$(TargetGroup)' == 'uap'">
+    <UAPToolsPackageVersion>1.0.5</UAPToolsPackageVersion>
```
Comment (dagood) [inline external/test-runtime/XUnit.Runtime.depproj]: While you're moving this, I think it should go in dependencies.props, in the `Full package version strings that are used in other parts of the build.` section. (No condition needed.) imo `*PackageVersion` properties belong there.

File: external/test-runtime/optional.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
     "net45": {
       "dependencies": {
         "Microsoft.DotNet.IBCMerge": "4.6.0-alpha-00001",
-        "Microsoft.DotNet.UAP.TestTools": "1.0.2",
+        //"Microsoft.DotNet.UAP.TestTools": "1.0.2",
```
Comment (dagood) [inline external/test-runtime/optional.json]: I'd add a comment explaining why, or just delete the line completely.

File: src/System.Private.ServiceModel/tests/Common/Infrastructure/xunit/WcfFactAttribute.cs [hunk 3]
```diff
@@ -9,7 +9,8 @@
 namespace Infrastructure.Common
 {
     // TODO: Use the common tests projects instead of hardcoding System.ServiceModel.Primitives here.
-    [XunitTestCaseDiscoverer("Infrastructure.Common.WcfFactDiscoverer", "System.ServiceModel.Primitives.Tests")]
+    [XunitTestCaseDiscoverer("Infrastructure.Common.WcfFactDiscoverer", "Infrastructure.Common")]
+    //[XunitTestCaseDiscoverer("Infrastructure.Common.WcfFactDiscoverer", "System.ServiceModel.Primitives.Tests")]
```
Comment (dagood) [inline src/System.Private.ServiceModel/tests/Common/Infrastructure/xunit/WcfFactAttribute.cs]: I think you should get rid of this line and the TODO--then this file will be unchanged by the BT2.0 effort.

File: src/System.Private.ServiceModel/tests/Common/Unit/UnitTests.Common.csproj [hunk 4]
```diff
@@ -7,23 +7,21 @@
...
   <ItemGroup>
     <Compile Include="**\*.cs" />
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="$(WcfDuplexPkgProj)" />
-    <ProjectReference Include="$(WcfHttpPkgProj)" />
-    <ProjectReference Include="$(WcfPrimitivesPkgProj)" />
+    <ProjectReference Include="$(WcfDuplexRefProj)" />
+    <ProjectReference Include="$(WcfHttpRefProj)" />
+    <ProjectReference Include="$(WcfPrimitivesRefProj)" />
```
Comment (dagood) [inline src/System.Private.ServiceModel/tests/Common/Unit/UnitTests.Common.csproj]: This seems reasonable, but I can't find any precedent. Test projects in CoreFX just reference each other, never back to the product.

File: src/System.ServiceModel.Security/tests/System.ServiceModel.Security.Tests.csproj [hunk 5]
```diff
@@ -20,13 +19,18 @@
...
+    <TargetingPackExclusions Include="System.ServiceModel.Primitives" />
+    <TargetingPackExclusions Include="System.ServiceModel.Duplex" />
+    <TargetingPackExclusions Include="System.ServiceModel.Http" />
+    <TargetingPackExclusions Include="System.ServiceModel.NetTcp" />
+    <TargetingPackExclusions Include="System.ServiceModel.Security" />
+    <ReferenceFromRuntime Include="System.Private.ServiceModel" />
+
+    <ProjectReference Include='$(WcfUnitTestCommonProj)'>
+      <Project>{E896294A-AB4A-4AF5-A01C-A19E3972EFF9}</Project>
+      <Name>UnitTests.Common</Name>
```
Comment (dagood) [inline src/System.ServiceModel.Security/tests/System.ServiceModel.Security.Tests.csproj]: Is including the guid and name necessary?

**Comment by dagood** [comment]
I applied https://github.com/dagood/wcf/commit/607bafa074326b16b51afc74256948419b2c48ec (References -> ProjectReferences for System.ServiceModel.Primitives) to get the build order to work itself out correctly, and after that, `.\build.cmd -framework=uap -tests` built and tested successfully from a clean `bin`. Really happy the common unit testing code turned out fine as projects!

File: src/System.Private.ServiceModel/tests/Common/Unit/UnitTests.Common.csproj (hunk 4 above)
Comment (StephenBonikowsky) [inline src/System.Private.ServiceModel/tests/Common/Unit/UnitTests.Common.csproj]: Yeah, hope we don't discover any unintended consequences.

**Comment by StephenBonikowsky** [comment]
Ok, I think it is ready to merge.

**Pull request merged**